dynamiq/connections/connections.py

```diff
@@ -932,6 +933,58 @@ def connect(self):
         pass
 
 
+class PGVector(BaseConnection):
+    type: Literal[ConnectionType.PGVector] = ConnectionType.PGVector
+    host: str = Field(default_factory=partial(get_env_var, "PGVECTOR_HOST", "localhost"))
+    port: str | int = Field(default_factory=partial(get_env_var, "PGVECTOR_PORT", "5432"))
+    database: str = Field(default_factory=partial(get_env_var, "PGVECTOR_DATABASE", "db"))
+    user: str = Field(default_factory=partial(get_env_var, "PGVECTOR_USER", "postgres"))
+    password: str = Field(default_factory=partial(get_env_var, "PGVECTOR_PASSWORD", "password"))
```

Comment:
What about `SSLMODE`?

Reply:
I think it uses `prefer` mode for `SSLMODE` by default (meaning `first try an SSL connection; if that fails, try a non-SSL connection`), but, if needed, I can add it to env vars and the connection string.

Reply:
Also, are there any more params we should add to the connection (I'm using [this documentation](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-PARAMKEYWORDS) for reference)?

Reply:
I think we already have the bare minimum.

Reply:
@delamainer Should I still add the `SSLMODE` parameter to the connection? It would mean that more environment variables will be needed to provide local paths to relevant SSL certificates. Also, it would be harder to construct the connection string as its form will depend on which option was selected for the `SSLMODE` param (basically, a couple of if-statements should be added in this case).

Reply:
Let's omit it.

Reply:
Note: update the environment variable names too.

Reply:
Meaning `PGVECTOR_HOST` ->`POSTGRESQL_HOST` and so on?

Reply:
Renamed the environment variables for the `PGVector` connection to `POSTGRES_...` in the `.env.example`, `connections.py`, and RAG example for `pgvector`.